src/core/tsdf_submap_collection.cpp

```diff
@@ -41,6 +41,35 @@ void TsdfSubmapCollection::createNewSubMap(const Transformation &T_M_S,
   TsdfSubmap::Ptr tsdf_sub_map(
       new TsdfSubmap(T_M_S, submap_id, tsdf_map_config_));
   tsdf_sub_maps_.push_back(tsdf_sub_map);
+  id_to_submap_.emplace(submap_id, tsdf_sub_map);
+}
+
+void TsdfSubmapCollection::createNewSubMap(const Transformation &T_M_S) {
+  // Increment ID and create new submap
+  SubmapID new_ID = 0;
+  if (!tsdf_sub_maps_.empty()) new_ID = tsdf_sub_maps_.back()->getID() + 1;
+  createNewSubMap(T_M_S, new_ID);
+}
+
+bool TsdfSubmapCollection::duplicateSubMap(const cblox::SubmapID source_submap_id, const cblox::SubmapID new_submap_id) {
```

Comment:
As above, I think this would be better implemented as a (deep) copy constructor for TSDF submaps. They do this in voxblox, for example here:

https://github.com/ethz-asl/voxblox/blob/master/voxblox/include/voxblox/core/layer.h#L49

and here:

https://github.com/ethz-asl/voxblox/blob/master/voxblox/include/voxblox/core/tsdf_map.h#L38

Lets do the same for tsdf submaps